### PR TITLE
Reformatting `nested_indices_where`

### PR DESCRIPTION
**Pull request by hirwa-nshuti**
This PR resolves issue #2356 

Added Functional examples

### CODE REVIEW

File: ivy/functional/ivy/nest.py [hunk 1]
```diff
@@ -315,7 +315,7 @@ def map_nest_at_indices(nest: Iterable, indices: Tuple, fn: Callable):
 
 
 def nested_indices_where(
-    nest: Iterable,
+    nest: Union[List, Tuple, Dict],
```
Comment (jkeane508) [inline ivy/functional/ivy/nest.py:318]: Is this change necessary? These are all encapsulated in Iterable

**Review by jkeane508 — CHANGES_REQUESTED** [review]
Hi Felix,

This looks good however I'm not sure the change for the type hint for dtype is necessary :)

Thanks,
James

**Comment by hirwa-nshuti** [comment]
Hi @jkeane508 I made the requested changes.

Thanks.

**Comment by jkeane508** [comment]
Looks good to me!
